osu.Game.Rulesets.Taiko/Edit/Blueprints/TaikoSpanPlacementBlueprint.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Graphics;
+using osu.Framework.Input.Events;
+using osu.Game.Rulesets.Edit;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Types;
+using osu.Game.Rulesets.Taiko.Objects;
+using osu.Game.Rulesets.Taiko.UI;
+using osuTK;
+using osuTK.Input;
+
+namespace osu.Game.Rulesets.Taiko.Edit.Blueprints
+{
+    public class TaikoSpanPlacementBlueprint : PlacementBlueprint
+    {
+        private readonly HitPiece headPiece;
+        private readonly HitPiece tailPiece;
+
+        private readonly LengthPiece lengthPiece;
+
+        private readonly IHasDuration spanPlacementObject;
+
+        public TaikoSpanPlacementBlueprint(HitObject hitObject)
+            : base(hitObject)
+
+        {
+            spanPlacementObject = hitObject as IHasDuration;
+
+            RelativeSizeAxes = Axes.Both;
+
+            InternalChildren = new Drawable[]
+            {
+                headPiece = new HitPiece
+                {
+                    Size = new Vector2(TaikoHitObject.DEFAULT_SIZE * TaikoPlayfield.DEFAULT_HEIGHT)
+                },
+                lengthPiece = new LengthPiece
+                {
+                    Height = TaikoHitObject.DEFAULT_SIZE * TaikoPlayfield.DEFAULT_HEIGHT
+                },
+                tailPiece = new HitPiece
+                {
+                    Size = new Vector2(TaikoHitObject.DEFAULT_SIZE * TaikoPlayfield.DEFAULT_HEIGHT)
+                }
+            };
+        }
+
+        private double originalStartTime;
+
+        protected override bool OnMouseDown(MouseDownEvent e)
+        {
+            if (e.Button != MouseButton.Left)
+                return false;
+
+            BeginPlacement(true);
+            return true;
+        }
+
+        protected override void OnMouseUp(MouseUpEvent e)
+        {
+            if (e.Button != MouseButton.Left)
+                return;
+
+            base.OnMouseUp(e);
+            EndPlacement(true);
+        }
+
+        public override void UpdatePosition(SnapResult result)
+        {
+            base.UpdatePosition(result);
+
+            if (PlacementActive)
+            {
+                if (result.Time is double endTime)
+                {
+                    if (endTime < originalStartTime)
+                    {
+                        HitObject.StartTime = endTime;
+                        spanPlacementObject.Duration = Math.Abs(endTime - originalStartTime);
+                        headPiece.Position = ToLocalSpace(result.ScreenSpacePosition);
+                        lengthPiece.X = headPiece.X;
+                        lengthPiece.Width = tailPiece.X - headPiece.X;
+                    }
+                    else
+                    {
+                        spanPlacementObject.Duration = Math.Abs(endTime - originalStartTime);
```

Comment:
This probably needs to also set `HitObject.StartTime = originalStartTime`?

Behaves kinda weirdly: https://drive.google.com/file/d/1RbHwwtKOxjwUOCGAI8KtXFK7mJjrt6E_/view